beacon-chain/forkchoice/doubly-linked-tree/last_root.go

```diff
@@ -0,0 +1,28 @@
+package doublylinkedtree
+
+import (
+	"github.com/prysmaticlabs/prysm/v4/consensus-types/primitives"
+	"github.com/prysmaticlabs/prysm/v4/time/slots"
+)
+
+// LastRoot returns the last canonical block root in the given epoch
+func (f *ForkChoice) LastRoot(epoch primitives.Epoch) [32]byte {
+	head := f.store.headNode
+	headEpoch := slots.ToEpoch(head.slot)
+	epochEnd, err := slots.EpochEnd(epoch)
+	if err != nil {
+		return [32]byte{}
+	}
+	if headEpoch < epoch {
+		return [32]byte{}
```

Comment:
Shouldn't we be returning errors here and above ? If the consumer can't get the correct root, we should return them an error rather than a zero-hash